PaintingTrivia.py

```diff
@@ -18,25 +32,77 @@ def set_session():
 def index():
     if 'username' in session:
         # return 'Logged in as %s' % Markup.escape(session['username'])
-        print 'index to show entries'
         return redirect(url_for('show_entries'))
     else:
-        print 'index to login'
-        return redirect(url_for('login'))
+        return redirect(url_for('menu'))
+
+
+@app.route('/main', methods=['GET', 'POST'])
+def menu():
+    if request.method == 'POST':
+        print 'Hey'
+        x = layout_buttons(request)
+        if x:
+            return x
+    return render_template('menu.html')
 
 
 @app.route('/login', methods=['GET', 'POST'])
 def login():
     if request.method == 'POST':
-        session['username'] = request.form['username']
-        return redirect(url_for('index'))
-    return render_template('menu.html')
+        x = layout_buttons(request)
+        if x:
+            return x
+        username_aux = request.form['username']
+        password_aux = request.form['password']
+        if not (username_aux == '' or password_aux == ''):
```

Comment:
prefer a shorter version and use weak typing since you are using python (https://docs.python.org/release/2.5.2/lib/truth.html)
```
if (username_aux and password_aux)
```